upload/upload.go

```diff
@@ -233,12 +235,35 @@ func Upload(args []string) error {
 	var files []string
 	var outFiles []string
 
+	// Shift flag and their arguments from the end to the beginning
+	// if more boolean flags are added in the future the following needs a slight modification
+	for k := len(args) - 1; k > 0; k-- {
+		if args[len(args)-1][0:1] != "-" && (args[len(args)-2][0:1] != "-" || args[len(args)-2] == "-r") {
+
+			break
+		}
+		args = append(args[0:1], append(args[len(args)-1:], args[1:len(args)-1]...)...)
+	}
+
 	err := Args.Parse(args[1:])
 	if err != nil {
 		return fmt.Errorf("failed parsing arguments, reason: %v", err)
 	}
 
-	// Check that we have a private key to decrypt with
+	// Check that specified upload directory is valid, i.e. not a filepath or a flag
+	info, err := os.Stat(*uploadDir)
+
+	// Dereference the pointer to a string
+	uploadDirString := ""
+	if uploadDir != nil {
+		uploadDirString = *uploadDir
+	}
+
+	if (!os.IsNotExist(err) && !info.IsDir()) || (uploadDirString != "" && uploadDirString[0:1] == "-") {
+		return errors.New(*uploadDir + " is not a valid upload directory")
+	}
+
```

Comment:
Nitpicking here, but isn't source directory the proper term?

Reply:
The above part refers to the `target` directory (though I now see how this can be confusing). I changed `upload` to `target` whenever possible so as to make var names and text clearer.